include/SDL3/SDL_audio.h

```diff
@@ -560,6 +551,28 @@ extern SDL_DECLSPEC const char *SDLCALL SDL_GetAudioDeviceName(SDL_AudioDeviceID
  */
 extern SDL_DECLSPEC int SDLCALL SDL_GetAudioDeviceFormat(SDL_AudioDeviceID devid, SDL_AudioSpec *spec, int *sample_frames);
 
+/**
+ * Get the current channel map of an audio device.
+ *
+ * Channel maps are optional; most things do not need them, instead passing
+ * data in the [order that SDL expects](CategoryAudio#channel-layouts).
+ *
+ * Audio devices usually have no remapping applied. This is represented by
+ * returning NULL, and does not signify an error.
+ *
+ * The returned array follows the SDL_GetStringRule (even though, strictly
+ * speaking, it isn't a string, it has the same memory manangement rules).
+ *
+ * \param devid the instance ID of the device to query.
+ * \returns an array of the current channel mapping, with as many elements as the current output spec's channels, or NULL if default.
```

Comment:
This should probably have an out parameter of the count, as otherwise there could be a race condition where the channel count is changed while processing this array.

Reply:
Fixed in 9a2d290a926087cfdabf6452fe33cc458445f120.